### PR TITLE
Refactor block producers interface

### PR DESCRIPTION
**Pull request by nholland94**
Currently, the way block producers are configured in terraform is overly specific to the nautical (whale/fish) stake distribution setup we tend to use on the testnet deploys. Now that we want to deploy networks using the same stack from our new integration test framework, we need to lift the nautical abstraction up a layer in terraform. This PR includes 3 things (separated into their own commits):

1) A refactor of the `kubernetes` module interface which generalizes the block producer configuration.
2) An example network configuration which uses this new interface (note that others will need to be ported to this format when this lands).
3) A refactor of the `block-producer` helm chart interface which matches the nested structure of the `snark-worker` interface, while making it easier to work with from the new `kubernetes` module interface.

None of these changes should change any behavior (except that we will need to update each of the testnet configurations). These changes have been partially tested already, but I am also deploying an instance of the example `pancake` network with a new build in order to check the health of the deploy fully.

### CODE REVIEW

**Comment by nholland94** [comment]
> btw think the UserAgent EnvVar declarations in the block-producer template could use a refactorization? Seems like checking for `""` and no defaults for these values seems a bit off.

Yeah, I thought about this too. Seemed off to me, then I looked and realized that it's because there are defaults in the script if they aren't set. But perhaps it's better to just always set it from our deployment pipeline?

**Comment by nholland94** [comment]
I had to make some additional changes after merging @yourbuddyconner's work for runtime configurations and kubernetes seeds in deployments. Most notably, I have lifted the `daemon.json` configuration completely out to the highest level so that we can pass one in directly for integration tests. If this is not desirable, we can have a conversation about how to support this workflow and the old one simultaneously.

**Comment by 0x0I** [comment]
> > btw think the UserAgent EnvVar declarations in the block-producer template could use a refactorization? Seems like checking for `""` and no defaults for these values seems a bit off.
> 
> Yeah, I thought about this too. Seemed off to me, then I looked and realized that it's because there are defaults in the script if they aren't set. But perhaps it's better to just always set it from our deployment pipeline?

Makes sense

> > btw think the UserAgent EnvVar declarations in the block-producer template could use a refactorization? Seems like checking for `""` and no defaults for these values seems a bit off.
> 
> Yeah, I thought about this too. Seemed off to me, then I looked and realized that it's because there are defaults in the script if they aren't set. But perhaps it's better to just always set it from our deployment pipeline?

Got it and falling back to defaults in the script or default deployment Helm values makes sense to me though I could see always setting it to be more explicit. I guess what's really throwing me off is the comparison to the empty string. Seems like a basic check against the value (false for nil or empty) would work.

File: terraform/modules/kubernetes/testnet/helm.tf [hunk 1]
```diff
@@ -14,82 +15,66 @@ provider helm {
...
     }
-    blockProducersWithBots    = var.fish_block_producers_with_bots
-    blockProducersWithPoints  = var.fish_block_producers_with_points
-    botsImage                 = var.coda_bots_image
-    pointsImage               = var.coda_points_image
-    faucetAmount              = var.coda_faucet_amount
-    faucetFee                 = var.coda_faucet_fee
+
+    blockProducerConfigs = [
+      for index, config in var.block_producer_configs: {
```
Comment (yourbuddyconner) [inline terraform/modules/kubernetes/testnet/helm.tf:63]: Whoa this is a killer feature, hadn't found the use for a for-loop in terraform yet...

**Pull request merged**